operations/alloy-mixin/dashboards/prometheus.libsonnet

```diff
@@ -430,63 +456,123 @@ local filename = 'alloy-prometheus-remote-write.json';
+        ||| % $._config,
 
   local componentQuery = 
     if std.isEmpty($._config.filterSelector) then
-    |||
-        label_values(prometheus_remote_write_wal_samples_appended_total{job=~"$job", instance=~"$instance", component_id=~"prometheus.remote_write.*"}, component_id)
-    |||
+      if $._config.enableAlloyCluster then
+        |||
+            label_values(prometheus_remote_write_wal_samples_appended_total{cluster_name=~"$alloyCluster", job=~"$job", instance=~"$instance", component_id=~"prometheus.remote_write.*"}, component_id)
+        |||
+      else
+        |||
+            label_values(prometheus_remote_write_wal_samples_appended_total{job=~"$job", instance=~"$instance", component_id=~"prometheus.remote_write.*"}, component_id)
+        |||    
     else
-    |||
-        label_values(prometheus_remote_write_wal_samples_appended_total{%(filterSelector)s, job=~"$job", instance=~"$instance", component_id=~"prometheus.remote_write.*"}, component_id)
-    ||| % $._config,
+      if $._config.enableAlloyCluster then
+        |||
+            label_values(prometheus_remote_write_wal_samples_appended_total{%(filterSelector)s, cluster_name=~"$alloyCluster", job=~"$job", instance=~"$instance", component_id=~"prometheus.remote_write.*"}, component_id)
+        ||| % $._config
+      else
+        |||
+            label_values(prometheus_remote_write_wal_samples_appended_total{%(filterSelector)s, job=~"$job", instance=~"$instance", component_id=~"prometheus.remote_write.*"}, component_id)
+        ||| % $._config,
 
   local urlQuery = 
     if std.isEmpty($._config.filterSelector) then
-    |||
-        label_values(prometheus_remote_storage_sent_batch_duration_seconds_sum{job="$job", instance=~"$instance", component_id=~"$component"}, url)
-    |||
+      if $._config.enableAlloyCluster then
+        |||
+            label_values(prometheus_remote_storage_sent_batch_duration_seconds_sum{cluster_name=~"$alloyCluster", job="$job", instance=~"$instance", component_id=~"$component"}, url)
+        |||
+      else
+        |||
+            label_values(prometheus_remote_storage_sent_batch_duration_seconds_sum{job="$job", instance=~"$instance", component_id=~"$component"}, url)
+        |||
     else
-    |||
-        label_values(prometheus_remote_storage_sent_batch_duration_seconds_sum{%(filterSelector)s, job="$job", instance=~"$instance", component_id=~"$component"}, url)
-    ||| % $._config,
+      if $._config.enableAlloyCluster then
+        |||
+            label_values(prometheus_remote_storage_sent_batch_duration_seconds_sum{%(filterSelector)s, cluster_name=~"$alloyCluster", job="$job", instance=~"$instance", component_id=~"$component"}, url)
+        ||| % $._config
+      else
+        |||
+            label_values(prometheus_remote_storage_sent_batch_duration_seconds_sum{%(filterSelector)s, job="$job", instance=~"$instance", component_id=~"$component"}, url)
+        ||| % $._config,
```

Comment:
all this branching is not scaling very well. If we add another variable it will just look awful. Isn't there a way to make this simpler? I'm not familiar with the dashboards config

Reply:
Will consult @v-zhuravlev on this, since he is a jsonnet sorcerer.